pkg/signup/keymanager_test.go

```diff
@@ -208,20 +196,26 @@ func TestKeyManagerKeyParsing(t *testing.T) {
 	})
 
 	t.Run("validate with invalid keys", func(t *testing.T) {
-		// check if the verification fails on wrong keys
+		// Create KeyManager instance.
+		configRegistry.GetViperInstance().Set("auth_client.public_keys_url", keysEndpointURL)
+		assert.Equal(t, configRegistry.GetAuthClientPublicKeysURL(), keysEndpointURL, "key url not set correctly for testing")
+		keyManager, err := signup.NewKeyManager(logger, configRegistry)
+
+		// check if the keys can be used to verify a JWT
 		var statictests = []struct {
-			name     string
-			jwt      string
-			kidToUse string
+			name string
+			jwt  string
+			kid  string
 		}{
-			// keys are swapped here to get invalid keys
-			{"JWT0", jwt0, keyKID1},
-			{"JWT1", jwt1, keyKID0},
+			{"valid JWT 0", jwt0, kid1},
+			{"valid JWT 1", jwt1, kid0},
 		}
 		for _, tt := range statictests {
 			t.Run(tt.name, func(t *testing.T) {
 				_, err = jwt.Parse(tt.jwt, func(token *jwt.Token) (interface{}, error) {
-					return keyManager.Key(tt.kidToUse)
+					kid := token.Header["kid"]
+					require.NotEqual(t, tt.kid, kid)
+					return keyManager.Key(tt.kid)
 				})
 				require.Error(t, err)
```

Comment:
Worth checking the error message too to make sure you got the expected token validation error here.

Reply:
Fixed 1786a4d.